torchtext/csrc/core/vocabulary.cpp

```diff
@@ -0,0 +1,37 @@
+#include "vocabulary.h"
+#include <iostream>
+
+namespace torchtext {
+namespace core {
+size_t Vocabulary::word_hash(const std::string& word) {
+  return std::hash<std::string>()(word);
```

Comment:
Do you know what hash function this uses? I'd prefer https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function

Reply:
No I don't but it can't be a bad one. Can it? I'll look into Fowler hash function.

Reply:
@cpuhrsch I looked into it and apparently GCC is using MurmurHashUnaligned2 (see [this](https://stackoverflow.com/questions/19411742/what-is-the-default-hash-function-used-in-c-stdunordered-map)). Maybe it would be better if we implemented the hash function ourselves to be sure that it will perform the same everywhere.

Reply:
Yes ideally we implement it ourselves, because this hash function also has particular properties for text.

Reply:
I did implement it.